lib/rubocop/config.rb

```diff
@@ -373,8 +373,23 @@ def file_to_include?(file)
       absolute_file_path = File.expand_path(file)
 
       patterns_to_include.any? do |pattern|
-        match_path?(pattern, relative_file_path) ||
-          match_path?(pattern, absolute_file_path)
+        if block_given?
+          yield pattern, relative_file_path, absolute_file_path
+        else
+          match_path?(pattern, relative_file_path) ||
+            match_path?(pattern, absolute_file_path)
+        end
+      end
+    end
+
+    def allowed_camel_case_file?(file)
+      file_to_include?(file) do |pattern, relative_path, absolute_path|
+        if pattern.to_s =~ /[A-Z]/
```

Comment:
I think this if statement can be converted to

```ruby
pattern.to_s =~ /[A-Z]/ &&
  (match_path?(pattern, relative_path) ||
   match_path?(pattern, absolute_path))
```
That way the else can be eliminated entirely. 

Also, what kind of object is `pattern`? Assuming that `pattern` is a symbol, `to_s` is unnecessary because `=~` works with symbols.

Reply:
> That way the else can be eliminated entirely.

Oh, I see! I fixed the conditional expression simple.

> Also, what kind of object is `pattern`?

If `!ruby/regexp /regexp$/` is written in `Include` of .rubocop.yml, a `Regexp` instance is returned.
https://github.com/rubocop-hq/rubocop/blob/1bff66b2b8bafd215c106ee06411277d5554581d/spec/rubocop/cli_spec.rb#L996

In that case this code need `to_s` to prevent `TypeError`.

Reply:
> If !ruby/regexp /regexp$/ is written in Include of .rubocop.yml, a Regexp instance is returned.

That makes more sense. I wasn't even thinking about `pattern` being regex. I'm so used to the paradigm of `to_s =~ //` being used unnecessarily with symbols.